app/controllers/groups_controller.rb

```diff
@@ -13,20 +13,21 @@ def show
   end
 
   def new
-    @group = Group.new
-    @neighborhoods = Neighborhood.all.map{|u| [ u.name, u.id ] }
-
     if current_user
-     @user = current_user
+      @group = Group.new
+      @neighborhoods = Neighborhood.all.map{|u| [ u.name, u.id ] }
     else
      redirect_to new_user_session_path, notice: 'You are not logged in.'
     end
   end
 
+
+  # When User creates group, create a UserGroup with is_leader: true
   def create
     @group = Group.new(group_params)
     respond_to do |format|
       if @group.save
+        user_group = UserGroup.create(user: current_user, group: @group, is_leader: true)
```

Comment:
I don't think you should be creating `UserGroups` manually. There should be a method like:
```ruby
@group.add_user(current_user, make_leader=true)
```